pkg/mover/intermediate.go

```diff
@@ -0,0 +1,106 @@
+// Copyright 2021 VMware, Inc.
+// SPDX-License-Identifier: BSD-2-Clause
+
+package mover
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/tarball"
+	"github.com/vmware-tanzu/asset-relocation-tool-for-kubernetes/internal"
+	"helm.sh/helm/v3/pkg/chart"
+)
+
+var (
+	// ErrNotIntermediateBundle when a verified path does not have expected intermediate bundle contents
+	ErrNotIntermediateBundle = errors.New("not an intermediate chart bundle")
+)
+
+func saveIntermediateBundle(cd *ChartData, tarFile string, log Logger) error {
```

Comment:
Could you add a comment explaining what this function will do, thanks!

Reply:
instead of passing the whole ChartData struct, wouldn't make more sense to pass the imageChanges?

Reply:
**ChartData** holds the **chart**, the **hints** and the **images**, which is **all the data we need to store**, the other 5 parameters are different, the target tarball and a logger. This means (data, target, logger)

I do not want to explode this 3 params into 5 (data1, data2, data3, target, logger) I think we lose more than we gain. Having too many parameters is a sign that you might not be doing something the proper way, or that there is a better way. I think this was a better way.

Also this chartData made us realize we are not modeling ImageChanges properly, even if that should NOT be fixed in this same PR.

Reply:
doc string added